### ISSUE TITLE
getCatalogItem and searchCatalogItems sales rank

### ISSUE THREAD

**Issue by steverobison** [jlevers/selling-partner-api#64]
1st Issue: Sales rank is not passed to the item result for either of these calls in the catalog API.

Cause: Amazon passes the data back in the json as 'ranks' not 'sales_ranks' even though sales_ranks is how they want it to be set with the included_data.

Fix:  
Line 124 on the Model/CatalogItem.php - change to.  
       'sales_ranks' => 'ranks',

That fixes it so rank data can be properly deserialized. 

This uncovered a 2nd issue.  

2nd Issue: The 'ranks' attribute in the Model/Catalog/ItemSalesRank.php should be 'value'.

Fix:
To fix this, I just replaced all 'ranks' with 'value' in the Model/Catalog/ItemSalesRank.php.

Amazon is fun with their buggy API and api docs.

**Comment by steverobison** [comment]
Here is a report of the issue on Amazon.

https://github.com/amzn/selling-partner-api-models/issues/139

**Comment by jlevers** [comment]
Thanks! I'll update the models I'm using for code generation and close this issue once I've regenerated the library and published.

**Comment by jlevers** [comment]
@steverobison, could you look at the `CatalogApi` and `OldCatalogApi` and let me know where these changes need to happen? The filenames have changed due to some updates to the models, and I'm no longer sure where the issues you were running into were occurring.

**Comment by steverobison** [comment]
Sure.  I'll take a look and get back to you later today.

**Comment by steverobison** [comment]
Model/Catalog/Item.php - line 124 change to.

'sales_ranks' => 'ranks',

Then Model/Catalog/ItemSalesRank.php I changed all occurrences of 'rank' to 'value'.

**Comment by jlevers** [comment]
Could you give me an example of an ASIN that should return sales rank values? I'm testing with `B07S3VKFKN` and not getting any sales rank info either before or after I make the changes you specified?

**Comment by steverobison** [comment]
B00CIFZL2M

**Comment by jlevers** [comment]
Just figured out the issue -- I haven't used this section of the API much, and forgot I had to pass `['salesRanks']` in the `$included_data` parameter.

To get this fix to persist in future versions of the library, I changed some values in the JSON model for the Catalog API. That caused a couple other (minor) changes):
* `getRanks()` replaces `getSalesRanks()` when pulling the array of `ItemSalesRank` objects out of the `Item` model
* `getValues()` replaces `getRanks()` when pulling the numeric sales rank of an item out of the `ItemSalesRank` model

These updates are available in `v.3.1.2`.
